guides/source/association_basics.md

```diff
@@ -1015,6 +1015,8 @@ end
 
 With this declaration, Rails will keep the cache value up to date, and then return that value in response to the `size` method.
 
+NOTE: Counter cache will not be updated if you change the parent model primary key's value, which means orphaned models will still count towards the counter.
+
```

Comment:
Instead of inserting a callout in the middle of the text, what about integrating it at the end of the section?  Something like:

```diff
diff --git a/guides/source/association_basics.md b/guides/source/association_basics.md
index eb17b2f306..14d5ca52ed 100644
--- a/guides/source/association_basics.md
+++ b/guides/source/association_basics.md
@@ -1108,7 +1108,15 @@ end
 NOTE: You only need to specify the `:counter_cache` option on the `belongs_to`
 side of the association.
 
-Counter cache columns are added to the containing model's list of read-only attributes through `attr_readonly`.
+Counter cache columns are added to the owner model's list of read-only
+attributes through `attr_readonly`.
+
+If for some reason you change the value of an owner model's primary key, and do
+not also update the foreign keys of the counted models, then the counter cache
+may have stale data. In other words, any orphaned models will still count
+towards the counter. To fix a stale counter cache, use [`reset_counters`][].
+
+[`reset_counters`]: https://api.rubyonrails.org/classes/ActiveRecord/CounterCache/ClassMethods.html#method-i-reset_counters
 
 ##### `:dependent`
```